server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDecider.java

```diff
@@ -553,6 +539,17 @@ private Decision earlyTerminate(Map<String, DiskUsage> usages) {
         return null;
     }
 
+    public static long getExpectedShardSize(ShardRouting shardRouting, long defaultSize, RoutingAllocation allocation) {
+        return DiskThresholdDecider.getExpectedShardSize(
+            shardRouting,
+            defaultSize,
+            allocation.clusterInfo(),
+            allocation.snapshotShardSizeInfo(),
+            allocation.metadata(),
+            allocation.routingTable()
+        );
+    }
```

Comment:
This and original method seems to be used outside of `DiskThresholdDecider` a lot. I wonder if we should move it to a dedicated component?